## 시현님 데이터 : df_1

### 임직원
- 임직원 : 설 연휴 포함한 데이터 성능이 더 좋음

In [1]:
import pandas as pd
df_1_y = pd.read_csv('https://raw.githubusercontent.com/ShineHop/BDA_CJTheMarket/main/cj_empY.csv')
df_1_y

scd  gender  age_grp  order_date  total_qty  total_amt  \
0     20230101963226       1        3           0          6  39.433806   
1     20230101963235       0        4           0          2  19.234566   
2     20230101963247       1        3           0          6  45.516855   
3     20230101963251       0        3           0          2  10.845621   
4     20230101963253       1        2           0          6  47.014742   
...              ...     ...      ...         ...        ...        ...   
4708  20230131216567       0        5           1          6  27.652922   
4709  20230131216610       1        3           1         12  81.994470   
4710  20230131216674       1        4           1          4  37.317843   
4711  20230131216734       1        5           1          3  29.500314   
4712  20230131216857       1        5           1          5  43.364081   

      event_product   amt/qty  prime_yn  ctg_0  ctg_1  ctg_2  ctg_3  
0                 2  7.609629         0      1      0      1      0  
1                 2  9.617283         1      1      1      0      0  
2                 0  8.070212         1      1      1      0      1  
3                 0  5.422811         1      0      0      0      1  
4                 0  8.445540         1      1      1      0      1  
...             ...       ...       ...    ...    ...    ...    ...  
4708              1  5.733350         1      1      0      0      0  
4709              6  8.342528         1      1      1      0      1  
4710              0  9.329461         0      0      1      0      0  
4711              3  9.833438         1      1      0      1      0  
4712              4  8.672816         1      1      1      1      1  

[4713 rows x 13 columns]

In [6]:
# X, y 나누기
X = df_1_y.drop('prime_yn', axis=1)
y = df_1_y['prime_yn']

In [7]:
# train, valid 데이터셋 나누기
from sklearn.model_selection import train_test_split
X_train, X_valid, y_train, y_valid = train_test_split(X, y, stratify=y, random_state=42)
print(X_train.shape, X_valid.shape, y_train.shape, y_valid.shape)

(3534, 12) (1179, 12) (3534,) (1179,)


In [9]:
# 그래디언트 부스팅
from sklearn.ensemble import GradientBoostingClassifier
from sklearn.metrics import accuracy_score
from sklearn.metrics import classification_report
from sklearn.metrics import f1_score

gradient_boosting = GradientBoostingClassifier()

# 모델 학습
gradient_boosting.fit(X_train, y_train)

# 예측
pred = gradient_boosting.predict(X_valid)

# 정확도 평가
accuracy = accuracy_score(y_valid, pred)
print("Accuracy:", accuracy)

# 평가
cfreport = classification_report(y_valid, pred)
print("분류예측 레포트:\n", cfreport)

f1 = f1_score(y_valid, pred)
print("f1 score:\n", f1)

Accuracy: 0.6581849024597116
분류예측 레포트:
               precision    recall  f1-score   support

           0       0.69      0.26      0.38       471
           1       0.65      0.92      0.76       708

    accuracy                           0.66      1179
   macro avg       0.67      0.59      0.57      1179
weighted avg       0.67      0.66      0.61      1179

f1 score:
 0.7644652250146113


In [10]:
from sklearn.model_selection import GridSearchCV

gradient_boosting = GradientBoostingClassifier()

param_grid = {
    'max_depth': range(3,10),
    'learning_rate': [0.1, 0.01, 0.001],
    'n_estimators': [100, 500, 1000]
}

# 그리드 서치
grid_search = GridSearchCV(estimator=gradient_boosting, param_grid=param_grid, cv=3, scoring='accuracy')
grid_search.fit(X_train, y_train)

# 최적의 파라미터
print("Best Parameters:", grid_search.best_params_)
print("Best Score:", grid_search.best_score_)

Best Parameters: {'learning_rate': 0.1, 'max_depth': 7, 'n_estimators': 500}
Best Score: 0.6395019807583475


In [11]:
# 테스트 데이터로 예측 수행
pred = grid_search.predict(X_valid)

# 정확도 평가
accuracy = accuracy_score(y_valid, pred)
print("Accuracy:", accuracy)

# 모델 예측 및 평가
from sklearn.metrics import classification_report
from sklearn.metrics import f1_score
cfreport = classification_report(y_valid, pred)
print("분류예측 레포트:\n", cfreport)

f1 = f1_score(y_valid, pred)
print("f1 score:\n", f1)

Accuracy: 0.6675148430873622
분류예측 레포트:
               precision    recall  f1-score   support

           0       0.60      0.49      0.54       471
           1       0.70      0.79      0.74       708

    accuracy                           0.67      1179
   macro avg       0.65      0.64      0.64      1179
weighted avg       0.66      0.67      0.66      1179

f1 score:
 0.740053050397878


### 비임직원
- 비임직원 : 설 연휴 제거한 데이터 성능이 더 좋음

In [13]:
df_1_n = pd.read_csv('https://raw.githubusercontent.com/ShineHop/BDA_CJTheMarket/main/cj_empN.csv')
df_1_n

scd  gender  age_grp  order_date  total_qty   total_amt  \
0     20230101963244       0        4           0          3   10.234373   
1     20230101963277       1        3           0          1    9.190036   
2     20230101963302       0        4           0         17   83.446280   
3     20230101963319       1        4           0         23   73.452653   
4     20230101963336       0        3           0         30  119.943496   
...              ...     ...      ...         ...        ...         ...   
5935  20230131216709       0        4           1          6   27.526347   
5936  20230131216756       0        4           1          3   18.314758   
5937  20230131216771       0        4           1          2    9.125762   
5938  20230131216842       1        3           1          1   10.176411   
5939  20230131216844       0        4           1          8   44.927942   

      event_product    amt/qty  prime_yn  ctg_0  ctg_1  ctg_2  ctg_3  
0                 0   3.411458         1      0      1      0      0  
1                 0   9.190036         1      0      0      1      0  
2                 0   7.574948         0      1      1      1      1  
3                 0   4.093077         0      1      1      1      0  
4                 0   6.208069         1      1      1      1      1  
...             ...        ...       ...    ...    ...    ...    ...  
5935              0   7.037744         1      1      0      0      1  
5936              0   6.796144         0      0      1      0      0  
5937              0   4.562881         1      1      0      0      0  
5938              1  10.176411         0      0      0      1      0  
5939              0   6.307187         0      0      1      0      1  

[5940 rows x 13 columns]

In [14]:
# X, y 나누기
X = df_1_n.drop('prime_yn', axis=1)
y = df_1_n['prime_yn']

In [15]:
# train, valid 데이터셋 나누기
from sklearn.model_selection import train_test_split
X_train, X_valid, y_train, y_valid = train_test_split(X, y, stratify=y, random_state=42)
print(X_train.shape, X_valid.shape, y_train.shape, y_valid.shape)

(4455, 12) (1485, 12) (4455,) (1485,)


In [16]:
import xgboost as xgb

# XGBoost
xgb = xgb.XGBClassifier()
xgb.fit(X_train, y_train)

# 테스트 데이터로 예측 수행
pred = xgb.predict(X_valid)

# 정확도 평가
accuracy = accuracy_score(y_valid, pred)
print("Accuracy:", accuracy)

Accuracy: 0.6404040404040404


In [18]:
# 모델 예측 및 평가
cfreport = classification_report(y_valid, pred)
print("분류예측 레포트:\n", cfreport)

f1 = f1_score(y_valid, pred)
print("f1 score:\n", f1)

분류예측 레포트:
               precision    recall  f1-score   support

           0       0.67      0.75      0.71       861
           1       0.59      0.49      0.54       624

    accuracy                           0.64      1485
   macro avg       0.63      0.62      0.62      1485
weighted avg       0.63      0.64      0.63      1485

f1 score:
 0.5356521739130435


In [20]:
from sklearn.model_selection import GridSearchCV
import xgboost as xgb

# XGBoost 모델 초기화
xgb_model = xgb.XGBClassifier()

param_grid = {
    'max_depth': range(3,10),
    'learning_rate': [0.1, 0.01, 0.001],
    'n_estimators': [100, 500, 1000]
}

# 그리드 서치
grid_search = GridSearchCV(estimator=xgb_model, param_grid=param_grid, cv=3, scoring='accuracy')
grid_search.fit(X_train, y_train)

# 최적의 파라미터
print("Best Parameters:", grid_search.best_params_)
print("Best Score:", grid_search.best_score_)

Best Parameters: {'learning_rate': 0.01, 'max_depth': 7, 'n_estimators': 500}
Best Score: 0.6278338945005612


In [21]:
# 테스트 데이터로 예측 수행
pred = grid_search.predict(X_valid)

# 정확도 평가
accuracy = accuracy_score(y_valid, pred)
print("Accuracy:", accuracy)

# 모델 예측 및 평가
from sklearn.metrics import classification_report
from sklearn.metrics import f1_score
cfreport = classification_report(y_valid, pred)
print("분류예측 레포트:\n", cfreport)

f1 = f1_score(y_valid, pred)
print("f1 score:\n", f1)

Accuracy: 0.6498316498316499
분류예측 레포트:
               precision    recall  f1-score   support

           0       0.66      0.82      0.73       861
           1       0.63      0.41      0.50       624

    accuracy                           0.65      1485
   macro avg       0.64      0.62      0.61      1485
weighted avg       0.65      0.65      0.63      1485

f1 score:
 0.49612403100775193


## 성연님 데이터 : df_2

### 임직원

In [22]:
df_2_y = pd.read_csv('https://raw.githubusercontent.com/ShineHop/BDA_CJTheMarket/main/cj_y_grp.csv')
df_2_y

scd  gender  age_grp  order_date  total_qty  total_amt  \
0     20230101963226       1        3           0          6  39.433806   
1     20230101963235       0        4           0          2  19.234566   
2     20230101963247       1        3           0          6  45.516855   
3     20230101963251       0        3           0          2  10.845621   
4     20230101963253       1        2           0          6  47.014742   
...              ...     ...      ...         ...        ...        ...   
4708  20230131216567       0        5           1          6  27.652922   
4709  20230131216610       1        3           1         12  81.994470   
4710  20230131216674       1        4           1          4  37.317843   
4711  20230131216734       1        5           1          3  29.500314   
4712  20230131216857       1        5           1          5  43.364081   

      event_product  num_category   amt/qty  prime_yn  most_product_0  \
0                 2             2  7.609629         0               0   
1                 2             2  9.617283         1               1   
2                 0             3  8.070212         1               1   
3                 0             1  5.422811         1               0   
4                 0             3  8.445540         1               0   
...             ...           ...       ...       ...             ...   
4708              1             2  5.733350         1               1   
4709              6             7  8.342528         1               1   
4710              0             2  9.329461         0               0   
4711              3             3  9.833438         1               1   
4712              4             4  8.672816         1               1   

      most_product_1  most_product_2  most_product_3  
0                  0               1               0  
1                  0               0               0  
2                  0               0               0  
3                  0               0               1  
4                  1               0               0  
...              ...             ...             ...  
4708               0               0               0  
4709               0               0               0  
4710               1               0               0  
4711               0               0               0  
4712               0               0               0  

[4713 rows x 14 columns]

In [23]:
# X, y 나누기
X = df_2_y.drop('prime_yn', axis=1)
y = df_2_y['prime_yn']

In [24]:
# train, valid 데이터셋 나누기
from sklearn.model_selection import train_test_split
X_train, X_valid, y_train, y_valid = train_test_split(X, y, stratify=y, random_state=42)
print(X_train.shape, X_valid.shape, y_train.shape, y_valid.shape)

(3534, 13) (1179, 13) (3534,) (1179,)


In [25]:
# 그래디언트 부스팅
from sklearn.ensemble import GradientBoostingClassifier
from sklearn.metrics import accuracy_score
from sklearn.metrics import classification_report
from sklearn.metrics import f1_score

gradient_boosting = GradientBoostingClassifier()

# 모델 학습
gradient_boosting.fit(X_train, y_train)

# 예측
pred = gradient_boosting.predict(X_valid)

# 정확도 평가
accuracy = accuracy_score(y_valid, pred)
print("Accuracy:", accuracy)

# 평가
cfreport = classification_report(y_valid, pred)
print("분류예측 레포트:\n", cfreport)

f1 = f1_score(y_valid, pred)
print("f1 score:\n", f1)

Accuracy: 0.6446140797285835
분류예측 레포트:
               precision    recall  f1-score   support

           0       0.65      0.24      0.35       471
           1       0.64      0.91      0.76       708

    accuracy                           0.64      1179
   macro avg       0.65      0.58      0.55      1179
weighted avg       0.65      0.64      0.59      1179

f1 score:
 0.7553998832457678


In [26]:
from sklearn.model_selection import GridSearchCV

gradient_boosting = GradientBoostingClassifier()

param_grid = {
    'max_depth': range(3,10),
    'learning_rate': [0.1, 0.01, 0.001],
    'n_estimators': [100, 500, 1000]
}

# 그리드 서치
grid_search = GridSearchCV(estimator=gradient_boosting, param_grid=param_grid, cv=3, scoring='accuracy')
grid_search.fit(X_train, y_train)

# 최적의 파라미터
print("Best Parameters:", grid_search.best_params_)
print("Best Score:", grid_search.best_score_)

Best Parameters: {'learning_rate': 0.1, 'max_depth': 9, 'n_estimators': 500}
Best Score: 0.6290322580645161


In [27]:
# 테스트 데이터로 예측 수행
pred = grid_search.predict(X_valid)

# 정확도 평가
accuracy = accuracy_score(y_valid, pred)
print("Accuracy:", accuracy)

# 모델 예측 및 평가
from sklearn.metrics import classification_report
from sklearn.metrics import f1_score
cfreport = classification_report(y_valid, pred)
print("분류예측 레포트:\n", cfreport)

f1 = f1_score(y_valid, pred)
print("f1 score:\n", f1)

Accuracy: 0.6505513146734521
분류예측 레포트:
               precision    recall  f1-score   support

           0       0.58      0.45      0.51       471
           1       0.68      0.78      0.73       708

    accuracy                           0.65      1179
   macro avg       0.63      0.62      0.62      1179
weighted avg       0.64      0.65      0.64      1179

f1 score:
 0.7293035479632063


### 비임직원

In [28]:
df_2_n = pd.read_csv('https://raw.githubusercontent.com/ShineHop/BDA_CJTheMarket/main/cj_n_grp.csv')
df_2_n

scd  gender  age_grp  order_date  total_qty   total_amt  \
0     20230101963244       0        4           0          3   10.234373   
1     20230101963277       1        3           0          1    9.190036   
2     20230101963302       0        4           0         17   83.446280   
3     20230101963319       1        4           0         23   73.452653   
4     20230101963336       0        3           0         30  119.943496   
...              ...     ...      ...         ...        ...         ...   
5935  20230131216709       0        4           1          6   27.526347   
5936  20230131216756       0        4           1          3   18.314758   
5937  20230131216771       0        4           1          2    9.125762   
5938  20230131216842       1        3           1          1   10.176411   
5939  20230131216844       0        4           1          8   44.927942   

      event_product  num_category    amt/qty  prime_yn  most_product_0  \
0                 0             1   3.411458         1               0   
1                 0             1   9.190036         1               0   
2                 0             7   7.574948         0               0   
3                 0             3   4.093077         0               1   
4                 0             7   6.208069         1               0   
...             ...           ...        ...       ...             ...   
5935              0             2   7.037744         1               1   
5936              0             2   6.796144         0               0   
5937              0             1   4.562881         1               1   
5938              1             1  10.176411         0               0   
5939              0             2   6.307187         0               0   

      most_product_1  most_product_2  most_product_3  
0                  1               0               0  
1                  0               1               0  
2                  0               0               1  
3                  0               0               0  
4                  1               0               0  
...              ...             ...             ...  
5935               0               0               0  
5936               1               0               0  
5937               0               0               0  
5938               0               1               0  
5939               0               0               1  

[5940 rows x 14 columns]

In [29]:
# X, y 나누기
X = df_2_n.drop('prime_yn', axis=1)
y = df_2_n['prime_yn']

In [30]:
# train, valid 데이터셋 나누기
from sklearn.model_selection import train_test_split
X_train, X_valid, y_train, y_valid = train_test_split(X, y, stratify=y, random_state=42)
print(X_train.shape, X_valid.shape, y_train.shape, y_valid.shape)

(4455, 13) (1485, 13) (4455,) (1485,)


In [31]:
import xgboost as xgb

# XGBoost
xgb = xgb.XGBClassifier()
xgb.fit(X_train, y_train)

# 테스트 데이터로 예측 수행
pred = xgb.predict(X_valid)

# 정확도 평가
accuracy = accuracy_score(y_valid, pred)
print("Accuracy:", accuracy)

Accuracy: 0.6356902356902356


In [32]:
# 모델 예측 및 평가
cfreport = classification_report(y_valid, pred)
print("분류예측 레포트:\n", cfreport)

f1 = f1_score(y_valid, pred)
print("f1 score:\n", f1)

분류예측 레포트:
               precision    recall  f1-score   support

           0       0.67      0.73      0.70       861
           1       0.58      0.50      0.54       624

    accuracy                           0.64      1485
   macro avg       0.62      0.62      0.62      1485
weighted avg       0.63      0.64      0.63      1485

f1 score:
 0.5380017079419299


In [33]:
from sklearn.model_selection import GridSearchCV
import xgboost as xgb

# XGBoost 모델 초기화
xgb_model = xgb.XGBClassifier()

param_grid = {
    'max_depth': range(3,10),
    'learning_rate': [0.1, 0.01, 0.001],
    'n_estimators': [100, 500, 1000]
}

# 그리드 서치
grid_search = GridSearchCV(estimator=xgb_model, param_grid=param_grid, cv=3, scoring='accuracy')
grid_search.fit(X_train, y_train)

# 최적의 파라미터
print("Best Parameters:", grid_search.best_params_)
print("Best Score:", grid_search.best_score_)

Best Parameters: {'learning_rate': 0.1, 'max_depth': 4, 'n_estimators': 500}
Best Score: 0.6356902356902356


In [34]:
# 테스트 데이터로 예측 수행
pred = grid_search.predict(X_valid)

# 정확도 평가
accuracy = accuracy_score(y_valid, pred)
print("Accuracy:", accuracy)

# 모델 예측 및 평가
from sklearn.metrics import classification_report
from sklearn.metrics import f1_score
cfreport = classification_report(y_valid, pred)
print("분류예측 레포트:\n", cfreport)

f1 = f1_score(y_valid, pred)
print("f1 score:\n", f1)

Accuracy: 0.635016835016835
분류예측 레포트:
               precision    recall  f1-score   support

           0       0.66      0.75      0.70       861
           1       0.58      0.48      0.52       624

    accuracy                           0.64      1485
   macro avg       0.62      0.61      0.61      1485
weighted avg       0.63      0.64      0.63      1485

f1 score:
 0.522887323943662


df_1
- 임직원 : Gradient Boosting
  1. accuracy 0.66, f1-score 0.77
  2. accuracy 0.67, f1-score 0.74

- 비임직원 : XGBoost
  1. accuracy 0.64, f1-score 0.54
  2. accuracy 0.65, f1-score 0.5

df_2
- 임직원 : Gradient Boosting
  1. accuracy 0.64, f1-score 0.76
  2. accuracy 0.65, f1-score 0.73

- 비임직원 : XGBoost
  1. accuracy 0.64, f1-score 0.54
  2. accuracy 0.64, f1-score 0.52